driver/src/main/com/mongodb/reactivestreams/client/gridfs/GridFSBucket.java

```diff
@@ -556,10 +621,234 @@
      * @return a publisher with a single element, representing the amount of data written
      * @mongodb.server.release 3.6
      * @since 1.7
+     * @deprecated prefer {@link GridFSBucket#downloadToPublisher(ClientSession, String, GridFSDownloadOptions)} instead
      */
+    @Deprecated
     Publisher<Long> downloadToStream(ClientSession clientSession, String filename, AsyncOutputStream destination,
                                      GridFSDownloadOptions options);
 
+    /**
+     * Uploads the contents of the given {@code AsyncInputStream} to a GridFS bucket.
```

Comment:
Replace: 

    AsyncInputStream => Publisher

in all Javadoc

Reply:
Done - copy pasta

Reply:
Sorry should have pointed out these as well:

`    Downloads the contents of the stored file specified by {@code id} and writes the contents to the {@code destination} {@code AsyncOutputStream}.
`

Also need changing